src/main/java/redis/clients/jedis/DefaultJedisSocketFactory.java

```diff
@@ -64,6 +70,25 @@ public DefaultJedisSocketFactory(HostAndPort hostAndPort, JedisClientConfig conf
     }
   }
 
+  private void connectToFirstSuccsefulHost(Socket socket, HostAndPort hostAndPort) throws IOException{
+    List<InetAddress> hosts = Arrays.asList(InetAddress.getAllByName(hostAndPort.getHost()));
+    Collections.shuffle(hosts);
+    List<Exception> exceptions = new ArrayList<Exception>();
+    boolean connected = false;
+    for (InetAddress host : hosts) {
+      try {
+        socket.connect(new InetSocketAddress(host.getHostAddress(), hostAndPort.getPort()), connectionTimeout);
+        connected = true;
+        break;
+      } catch (Exception e) {
+        exceptions.add(e);
+      }
+    }
+    if (!connected) {
+      throw new ConnectException(exceptions.toString());
```

Comment:
```suggestion
      throw new Exception(exceptions.toString());
```

Reply:
@yangbodong22011 I disagree with this change for 2 reasons:
1. Some unit tests depend on the exception thrown here to be a ConnectException
2. I think its better to be specific when we can.
WDYT?

Reply:
IMO, it should be 
```
JedisConnectionException jce = JedisConnectionException("PROPER_MESSAGE");
for each exceptions => jce.addSuppressed(e);
throw jce;
```
then in createSocket()
```
if (ex instanceof JedisConnectionException) throw ex;
else throw ...;
```

Reply:
@adiamzn Agree with @sazzad16, pls refer to his opinion.

Reply:
@yangbodong22011 @sazzad16 
Thanks for the feedback!
Agreed, making the changes